# Bundesliga Home Advantage during COVID-19

## Research Question

Did the absence of spectators during the COVID-19 pandemic reduce the home advantage in the German Bundesliga?

This project analyses Bundesliga match data from multiple seasons before and during COVID-19.

## 1. Import Libaries

In [5]:
import pandas as pd
import matplotlib.pyplot as plt


Matplotlib is building the font cache; this may take a moment.


## 2. Load Example Dataset

In this section, the Bundesliga dataset for the 2018/19 season is loaded into a pandas DataFrame.

In [8]:
df_1819 = pd.read_csv("data/D1_1819.csv")

## 3. Explore the Dataset

Before starting the analysis, it is important to understand the structure of the dataset.

In this section, we examine the first rows of the dataset, identify the available variables, and inspect the dimensions and data types. This helps to verify that the dataset has been loaded correctly and provides an overview of the available information.

In [14]:
df_1819.head()



,Div,Date,HomeTeam,AwayTeam,FTHG,FTAG,FTR,HTHG,HTAG,HTR,...,BbAv<2.5,BbAH,BbAHh,BbMxAHH,BbAvAHH,BbMxAHA,BbAvAHA,PSCH,PSCD,PSCA
0,D1,24/08/2018,Bayern Munich,Hoffenheim,3,1,H,1,0,H,...,3.55,22,-2.00,1.92,1.87,2.05,1.99,1.23,7.15,14.10
1,D1,25/08/2018,Fortuna Dusseldorf,Augsburg,1,2,A,1,0,H,...,1.76,20,0.00,1.80,1.76,2.17,2.11,2.74,3.33,2.78
2,D1,25/08/2018,Freiburg,Ein Frankfurt,0,2,A,0,1,A,...,1.69,20,-0.25,2.02,1.99,1.92,1.88,2.52,3.30,3.07
3,D1,25/08/2018,Hertha,Nurnberg,1,0,H,1,0,H,...,1.76,20,-0.25,1.78,1.74,2.21,2.14,1.79,3.61,5.21
4,D1,25/08/2018,M'gladbach,Leverkusen,2,0,H,0,0,D,...,2.32,20,0.00,2.13,2.07,1.84,1.78,2.63,3.70,2.69


The following output displays the first five matches of the dataset.

In [ ]:
df_1819.columns



61

In [31]:
df_1819.shape

(306, 61)

The dataset contains 306 matches and 61 variables, confirming that the complete Bundesliga season 2018/19 has been loaded successfully.

In [32]:
df_1819.info()

<class 'pandas.DataFrame'>
RangeIndex: 306 entries, 0 to 305
Data columns (total 61 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Div       306 non-null    str    
 1   Date      306 non-null    str    
 2   HomeTeam  306 non-null    str    
 3   AwayTeam  306 non-null    str    
 4   FTHG      306 non-null    int64  
 5   FTAG      306 non-null    int64  
 6   FTR       306 non-null    str    
 7   HTHG      306 non-null    int64  
 8   HTAG      306 non-null    int64  
 9   HTR       306 non-null    str    
 10  HS        306 non-null    int64  
 11  AS        306 non-null    int64  
 12  HST       306 non-null    int64  
 13  AST       306 non-null    int64  
 14  HF        306 non-null    int64  
 15  AF        306 non-null    int64  
 16  HC        306 non-null    int64  
 17  AC        306 non-null    int64  
 18  HY        306 non-null    int64  
 19  AY        306 non-null    int64  
 20  HR        306 non-null    int64  
 21  AR  

## 4. Select Relevant Variables

The original dataset contains 61 variables. However, only a subset of these variables is required to analyse the home advantage in the Bundesliga.

For this project, we focus on:
- Date
- HomeTeam
- AwayTeam
- Full Time Home Goals (FTHG)
- Full Time Away Goals (FTAG)
- Full Time Result (FTR)

In [35]:
df_1819[["Date","HomeTeam","AwayTeam","FTHG","FTAG","FTR"]].head()


,Date,HomeTeam,AwayTeam,FTHG,FTAG,FTR
0,24/08/2018,Bayern Munich,Hoffenheim,3,1,H
1,25/08/2018,Fortuna Dusseldorf,Augsburg,1,2,A
2,25/08/2018,Freiburg,Ein Frankfurt,0,2,A
3,25/08/2018,Hertha,Nurnberg,1,0,H
4,25/08/2018,M'gladbach,Leverkusen,2,0,H


In [37]:
relevant_df = df_1819[
    ["Date", "HomeTeam", "AwayTeam", "FTHG", "FTAG", "FTR"]
]

In [38]:
relevant_df.head()

,Date,HomeTeam,AwayTeam,FTHG,FTAG,FTR
0,24/08/2018,Bayern Munich,Hoffenheim,3,1,H
1,25/08/2018,Fortuna Dusseldorf,Augsburg,1,2,A
2,25/08/2018,Freiburg,Ein Frankfurt,0,2,A
3,25/08/2018,Hertha,Nurnberg,1,0,H
4,25/08/2018,M'gladbach,Leverkusen,2,0,H


## 5. First Descriptive Statistics

To obtain an initial overview of the match outcomes, we calculate the frequency of home wins, away wins, and draws.

These descriptive statistics provide a first indication of the distribution of results before comparing different seasons.

In [39]:
relevant_df["FTR"].value_counts()

FTR
H    138
A     95
D     73
Name: count, dtype: int64

In [40]:
relevant_df["FTR"].value_counts(normalize=True) * 100

FTR
H    45.098039
A    31.045752
D    23.856209
Name: proportion, dtype: float64

### Interpretation

The results show that home wins occur more frequently than away wins during the 2018/19 Bundesliga season. This observation supports the assumption of a home advantage and serves as a baseline for later comparisons with the COVID-19 seasons.

# 6. Load Additional Seasons

To analyse changes in home advantage over time, additional Bundesliga seasons are loaded into the project.

In [43]:
df_1819 = pd.read_csv("data/D1_1819.csv")
df_1920 = pd.read_csv("data/D1_1920.csv")
df_2021 = pd.read_csv("data/D1_2021.csv")
df_2122 = pd.read_csv("data/D1_2122.csv")
df_2223 = pd.read_csv("data/D1_2223.csv")
df_2324 = pd.read_csv("data/D1_2324.csv")
df_2425 = pd.read_csv("data/D1_2425.csv")


## 7. Verify Additional Datasets

The 2018/19 season has already been explored in detail. Therefore, only the newly added datasets are checked to ensure that they have been loaded correctly and contain the expected number of matches.

Before combining the different seasons, we check the size of each dataset. A complete Bundesliga season with 18 teams should contain 306 matches.

In [44]:
df_1920.shape

(306, 105)

In [45]:
df_2021.shape

(306, 105)

In [46]:
df_2122.shape

(306, 105)

In [47]:
df_2223.shape


(306, 105)

In [48]:
df_2324.shape

(306, 105)

In [49]:
df_2425.shape

(306, 119)

In [50]:
print(df_1819.columns)
print(df_2425.columns)

Index(['Div', 'Date', 'HomeTeam', 'AwayTeam', 'FTHG', 'FTAG', 'FTR', 'HTHG',
       'HTAG', 'HTR', 'HS', 'AS', 'HST', 'AST', 'HF', 'AF', 'HC', 'AC', 'HY',
       'AY', 'HR', 'AR', 'B365H', 'B365D', 'B365A', 'BWH', 'BWD', 'BWA', 'IWH',
       'IWD', 'IWA', 'PSH', 'PSD', 'PSA', 'WHH', 'WHD', 'WHA', 'VCH', 'VCD',
       'VCA', 'Bb1X2', 'BbMxH', 'BbAvH', 'BbMxD', 'BbAvD', 'BbMxA', 'BbAvA',
       'BbOU', 'BbMx>2.5', 'BbAv>2.5', 'BbMx<2.5', 'BbAv<2.5', 'BbAH', 'BbAHh',
       'BbMxAHH', 'BbAvAHH', 'BbMxAHA', 'BbAvAHA', 'PSCH', 'PSCD', 'PSCA'],
      dtype='str')
Index(['Div', 'Date', 'Time', 'HomeTeam', 'AwayTeam', 'FTHG', 'FTAG', 'FTR',
       'HTHG', 'HTAG',
       ...
       'B365CAHH', 'B365CAHA', 'PCAHH', 'PCAHA', 'MaxCAHH', 'MaxCAHA',
       'AvgCAHH', 'AvgCAHA', 'BFECAHH', 'BFECAHA'],
      dtype='str', length=119)


### Observation

The datasets contain the same number of matches (306) but differ in the number of variables. Over the years, additional information such as playing time and match statistics has been added to the datasets.

Since this project only requires the variables *Date*, *HomeTeam*, *AwayTeam*, *FTHG*, *FTAG* and *FTR*, these differences do not affect the analysis.

## 8. Add Season Information

Before combining the datasets, a new variable called *Season* is added to each dataset.

This variable preserves the origin of every match after all seasons have been merged into a single dataset and allows comparisons between different seasons.

In [52]:
df_1819["Season"] = "2018/19"
df_1920["Season"] = "2019/20"
df_2021["Season"] = "2020/21"
df_2122["Season"] = "2021/22"
df_2223["Season"] = "2022/23"
df_2324["Season"] = "2023/24"
df_2425["Season"] = "2024/25"

/var/folders/t9/s5jgd_1x47vfy9jqd6dwm_m00000gn/T/ipykernel_5676/1485146738.py:7: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df_2425["Season"] = "2024/25"


In [53]:
df_1819.head()

,Div,Date,HomeTeam,AwayTeam,FTHG,FTAG,FTR,HTHG,HTAG,HTR,...,BbAH,BbAHh,BbMxAHH,BbAvAHH,BbMxAHA,BbAvAHA,PSCH,PSCD,PSCA,Season
0,D1,24/08/2018,Bayern Munich,Hoffenheim,3,1,H,1,0,H,...,22,-2.00,1.92,1.87,2.05,1.99,1.23,7.15,14.10,2018/19
1,D1,25/08/2018,Fortuna Dusseldorf,Augsburg,1,2,A,1,0,H,...,20,0.00,1.80,1.76,2.17,2.11,2.74,3.33,2.78,2018/19
2,D1,25/08/2018,Freiburg,Ein Frankfurt,0,2,A,0,1,A,...,20,-0.25,2.02,1.99,1.92,1.88,2.52,3.30,3.07,2018/19
3,D1,25/08/2018,Hertha,Nurnberg,1,0,H,1,0,H,...,20,-0.25,1.78,1.74,2.21,2.14,1.79,3.61,5.21,2018/19
4,D1,25/08/2018,M'gladbach,Leverkusen,2,0,H,0,0,D,...,20,0.00,2.13,2.07,1.84,1.78,2.63,3.70,2.69,2018/19


In [59]:
relevant_1819 = df_1819[
    ["Date", "HomeTeam", "AwayTeam", "FTHG", "FTAG", "FTR", "Season"]
]

relevant_1920 = df_1920[
    ["Date", "HomeTeam", "AwayTeam", "FTHG", "FTAG", "FTR", "Season"]
]

relevant_2021 = df_2021[
    ["Date", "HomeTeam", "AwayTeam", "FTHG", "FTAG", "FTR", "Season"]
]

relevant_2122 = df_2122[
    ["Date", "HomeTeam", "AwayTeam", "FTHG", "FTAG", "FTR", "Season"]
]

relevant_2223 = df_2223[
    ["Date", "HomeTeam", "AwayTeam", "FTHG", "FTAG", "FTR", "Season"]
]

relevant_2324 = df_2324[
    ["Date", "HomeTeam", "AwayTeam", "FTHG", "FTAG", "FTR", "Season"]
]

relevant_2425 = df_2425[
    ["Date", "HomeTeam", "AwayTeam", "FTHG", "FTAG", "FTR", "Season"]
]

## 9. Combine the Datasets

After selecting the relevant variables, all seasons are merged into a single DataFrame.

This combined dataset forms the basis for the subsequent analysis of the home advantage before, during and after the COVID-19 period.

In [61]:
df_all = pd.concat(
    [
        relevant_1819,
        relevant_1920,
        relevant_2021,
        relevant_2122,
        relevant_2223,
        relevant_2324,
        relevant_2425
    ],
    ignore_index=True
)


In [62]:
df_all["Season"].value_counts()

Season
2018/19    306
2019/20    306
2020/21    306
2021/22    306
2022/23    306
2023/24    306
2024/25    306
Name: count, dtype: int64

## 10. Explore the Combined Dataset

After combining all seasons into a single DataFrame, the structure of the new dataset is examined.

This step ensures that all matches have been merged correctly and that the dataset is ready for further analysis.

In [63]:
df_all.head()

,Date,HomeTeam,AwayTeam,FTHG,FTAG,FTR,Season
0,24/08/2018,Bayern Munich,Hoffenheim,3,1,H,2018/19
1,25/08/2018,Fortuna Dusseldorf,Augsburg,1,2,A,2018/19
2,25/08/2018,Freiburg,Ein Frankfurt,0,2,A,2018/19
3,25/08/2018,Hertha,Nurnberg,1,0,H,2018/19
4,25/08/2018,M'gladbach,Leverkusen,2,0,H,2018/19


In [65]:
df_all["Season"].value_counts()


Season
2018/19    306
2019/20    306
2020/21    306
2021/22    306
2022/23    306
2023/24    306
2024/25    306
Name: count, dtype: int64

## 11. Define the Analysis Periods

## 12. Compare Match Outcomes
The distribution of home wins, away wins and draws will be compared across the defined periods.

## 13. Data Visualisation
The results will be visualised using different charts to illustrate potential changes in the home advantage.

## 14. Statistical Analysis
Statistical tests will be used to determine whether the observed differences are statistically significant.

## 15. Discussion
The findings will be interpreted and discussed in the context of previous research on the home advantage.